src/acquisition/covid_hosp/common/test_utils.py

```diff
@@ -9,11 +9,53 @@
 """
 
 # standard library
+import unittest
 from pathlib import Path
+from unittest.mock import patch
 
 # third party
 import pandas
 
+from delphi.epidata.acquisition.covid_hosp.common.database import Database
+from delphi.epidata.client.delphi_epidata import Epidata
+from delphi.epidata.common.covid_hosp.covid_hosp_schema_io import CovidHospSomething
+import delphi.operations.secrets as secrets
+
+class TestDatabase(Database):
+  DATASET_NAME = 'mock_dataset'
+
+  @staticmethod
+  def create_mock_database(table_name=None,
+                           dataset_id=None,
+                           metadata_id=None,
+                           issue_col=None,
+                           csv_cols=[],
+                           key_cols=[],
+                           aggregate_cols=[]):
+    with patch.object(CovidHospSomething, 'get_ds_table_name', return_value=table_name), \
+        patch.object(CovidHospSomething, 'get_ds_dataset_id', return_value=dataset_id), \
+        patch.object(CovidHospSomething, 'get_ds_metadata_id', return_value=metadata_id), \
+        patch.object(CovidHospSomething, 'get_ds_issue_column', return_value=issue_col), \
+        patch.object(CovidHospSomething, 'get_ds_ordered_csv_cols', return_value=csv_cols), \
+        patch.object(CovidHospSomething, 'get_ds_key_cols', return_value=key_cols), \
+        patch.object(CovidHospSomething, 'get_ds_aggregate_key_cols', return_value=aggregate_cols):
+      return TestDatabase()
+
+class CovidHospTestCase(unittest.TestCase):
+
+  def setUp(self, database, tables):
+    # use the local instance of the Epidata API
+    Epidata.BASE_URL = 'http://delphi_web_epidata/epidata/api.php'
+
+    # use the local instance of the epidata database
+    secrets.db.host = 'delphi_database_epidata'
+    secrets.db.epi = ('user', 'pass')
+
+    # clear relevant tables
+    with database.connect() as db:
+      with db.new_cursor() as cur:
+        for table in tables:
+          cur.execute(f'truncate table {table}')
```

Comment:
the idea of refactoring into a superclassed `setUp()` is to reduce repetition as much as possible, and to use reflection-like methods instead of explicitly naming resources already associated somewhere else...  with these suggested changes, each subclass of `CovidHospTestCase` just needs to note the specific `Database` class under test, its file location for test data context (though there are other package-name-based ways to do this if youd like to explore), and any extra table(s) it might use (which ultimately can be removed).

```suggestion
class CovidHospTestCase(unittest.TestCase):

  # assign these in subclasses:
  db_class = None
  test_util_context = None
  # optionally extend or recreate this in subclasses:
  extra_tables_used = []

  def setUp(self):
    # use the local instance of the Epidata API
    Epidata.BASE_URL = 'http://delphi_web_epidata/epidata/api.php'

    # use the local instance of the epidata database
    secrets.db.host = 'delphi_database_epidata'
    secrets.db.epi = ('user', 'pass')
    
    # configure test data
    self.test_utils = UnitTestUtils(self.test_util_context)

    chs = CovidHospSomething()
    ds_name = db_class.DATASET_NAME

    # clear all relevant tables
    tables_used = ['covid_hosp_meta', chs.get_ds_table_name(ds_name)]
    if chs.get_ds_aggregate_key_cols(ds_name):
      tables_used.append(chs.get_ds_table_name(ds_name)+'_key')
    tables_used += self.extra_tables_used
    with self.db_class().connect() as db:
      with db.new_cursor() as cur:
        for table in tables_used:
          cur.execute(f'truncate table {table}')
```